data-structures/src/data/structure/linkedlist/LinkedList.java

```diff
@@ -0,0 +1,239 @@
+
+        requestedPlace.previous.next = requiredNode;
+        requiredNode.previous = requestedPlace.previous;
+        requiredNode.next = requestedPlace;
+        requestedPlace.previous = requiredNode;
+    }
+
+    public void addAll(LinkedList<E> other) {
+        requireNonNull(other, "The list must not be null!");
+
+        LinkedList<E> copy = makeCopyList(other);
+
+        tail.next = copy.head;
+        copy.head.previous = tail;
+        tail = copy.tail;
+        size += copy.size;
+    }
+
+    public void addAll(int index, LinkedList<E> other) {
+        requireNonNull(other, "The list must not be null!");
+
+        if (index > size && index < 0) {
+            throw new IllegalArgumentException("Enter a proper index value!");
+        }
+
+        if (index == size) {
+            addAll(other);
+            return;
+        }
+
+        LinkedList<E> copy = makeCopyList(other);
+        Node requestedPlace = findNode(index);
+
+        requestedPlace.previous.next = copy.head;
+        copy.head.previous = requestedPlace.previous;
+        copy.tail.next = requestedPlace;
+        requestedPlace.previous = copy.tail;
+        size += copy.size;
+    }
+
+    public E set(int index, E element) {
+        requireNonNull(element, "The element must not be null!");
+
+        if (index >= size && index < 0) {
+            throw new IllegalArgumentException("Enter a proper index value!");
+        }
+
+        Node requestedPlace = findNode(index);
+
+        final E replacedElement = requestedPlace.data;
+        requestedPlace.data = element;
+
+        return replacedElement;
+    }
+
+    public void clear() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
+    public E remove(int index) {
+        if (index >= size && index < 0) {
+            throw new IllegalArgumentException("Enter a proper index value!");
+        }
+
+        Node requestedPlace = findNode(index);
+
+        if (size == 1) {
+            clear();
+            return requestedPlace.data;
+        }
+
+        if (index == 0) {
+            requestedPlace.next.previous = null;
+            head = requestedPlace.next;
+            size--;
+            return requestedPlace.data;
+        }
+
+        if (index == size - 1) {
+            requestedPlace.previous.next = null;
+            tail = requestedPlace.previous;
+            size--;
+            return requestedPlace.data;
+        }
+
+        requestedPlace.previous.next = requestedPlace.next;
+        requestedPlace.next.previous = requestedPlace.previous;
+        size--;
+        return requestedPlace.data;
+    }
+
+    public boolean contains(E element) {
+        requireNonNull(element, "The element must not be null!");
+
+        Node temp = head;
+
+        for (int i = 0; i < size; i++) {
+            if (temp.data.equals(element)) {
+                return true;
+            }
+
+            temp = temp.next;
+        }

```

Comment:
There's no need for integers here. Check out this one:

~~~~Java
for (Node node = head; nonNull(node); node = node.next) {
  if (node.data.equals(element) {
    return true;
  }
}
~~~~

Cool, huh? :)

You can use this loop pattern in other methods as well.